feature-libs/cart/base/components/added-to-cart-toast/added-to-cart-toast.component.ts

```diff
@@ -0,0 +1,192 @@
+import {
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  HostBinding,
+  OnDestroy,
+  Renderer2,
+} from '@angular/core';
+import { WindowRef } from '@spartacus/core';
+import { Observable, OperatorFunction, Subscription } from 'rxjs';
+import { filter, take } from 'rxjs/operators';
+import {
+  ActiveCartFacade,
+  CartToastItem,
+  CartUiEventAddToCart,
+  CART_TOAST_STATE,
+  OrderEntry,
+} from '../../root';
+import { AddedToCartToastComponentService } from './added-to-cart-toast-component.service';
+
+@Component({
+  selector: 'cx-added-to-cart-toast',
+  templateUrl: './added-to-cart-toast.component.html',
+})
+export class AddedToCartToastComponent implements OnDestroy {
+  protected subscription = new Subscription();
+
+  customClass?: string;
+
+  headerElement: HTMLElement | null;
+
+  toastContainerClass: string;
+
+  timeout: number;
+
+  toastContainerBaseClass = 'toast-container';
+
+  scrollEventUnlistener: () => void;
+
+  @HostBinding('className') baseClass: string;
+
+  cartToasts$: Observable<CartToastItem[]> =
+    this.addedToCartToastService.getToasts();
+
+  constructor(
+    protected winRef: WindowRef,
+    protected renderer: Renderer2,
+    protected cd: ChangeDetectorRef,
+    protected addedToCartToastService: AddedToCartToastComponentService,
+    protected activeCartService: ActiveCartFacade,
+    protected el: ElementRef
+  ) {}
+
+  init(): void {
+    if (!this.customClass) {
+      this.customClass = 'cx-added-to-cart-toast';
+    }
+
+    this.baseClass = `${this.customClass}`;
+    this.toastContainerClass = this.toastContainerBaseClass;
+    this.headerElement = this.winRef.document.querySelector('header');
+  }
+
+  ngOnDestroy(): void {
+    this.subscription.unsubscribe();
+  }
+
+  addToast(event: CartUiEventAddToCart) {
+    this.setOffsetHeight();
+
+    this.activeCartService
+      .getLastEntry(event.productCode)
+      .pipe(
+        filter(Boolean) as OperatorFunction<OrderEntry | undefined, OrderEntry>,
+        take(1)
+      )
+      .subscribe((entries) => {
+        let toastContainerClass = `${this.toastContainerBaseClass} `;
```

Comment:
could `toastContainerClass` be a const?  If so, perhaps it can be not used at all if the string is used on once place.